src/ide/formatter.rs

```diff
@@ -1,18 +1,23 @@
 use cairo_lang_filesystem::db::FilesGroup;
-use cairo_lang_formatter::{FormatterConfig, get_formatted_file};
+use cairo_lang_formatter::get_formatted_file;
 use cairo_lang_parser::db::ParserGroup;
 use cairo_lang_utils::Upcast;
 use lsp_types::{DocumentFormattingParams, Position, Range, TextEdit};
 use tracing::error;
 
-use crate::lang::db::AnalysisDatabase;
 use crate::lang::lsp::LsProtoGroup;
+use crate::state::StateSnapshot;
 
 /// Format a whole document.
-pub fn format(params: DocumentFormattingParams, db: &AnalysisDatabase) -> Option<Vec<TextEdit>> {
+pub fn format(params: DocumentFormattingParams, state: StateSnapshot) -> Option<Vec<TextEdit>> {
+    let db = &*state.db;
     let file_uri = params.text_document.uri;
     let file = db.file_for_url(&file_uri)?;
 
+    let path = file_uri.path();
```

Comment:
you should rather use https://docs.rs/url/latest/url/struct.Url.html#method.to_file_path here